### ISSUE TITLE
Add version information to "bat cache" files to prevent incompatibilities

### ISSUE THREAD

**Issue by sharkdp** [sharkdp/bat#882]
There are numerous reports of bugs that emerged from cached assets that were generated with an older version of `bat` (#425, #853, #797, maybe #704).

One way to prevent this would be to store version information with the cached assets. This way, we could check for compatibility upon loading.

The version information should probably be stored in a separate text file with a very simple format which is unlikely to change in the future. Maybe just a simple `assets.yaml` file with:
``` yaml
version: "7.4.0"
```

**Comment by eth-p** [comment]
A couple of small considerations:

1. What should happen with a missing version file? Treat it as an unknown and incompatible syntax/theme set?

2. What should we do if the user has an incompatible syntax/theme set?

My thoughts on #2:

I don't think deleting it automatically would the correct course of action. If users update bat and find their theme or syntax mappings missing without any explanation, they might consider it a bug.

Ruling that out would leave us with two options: Warn the user and proceed anyways; or print an error and exit.

Personally, I prefer printing a stern warning and continuing. Let the user proceed at their own risk knowing the potential consequences.

**Comment by sharkdp** [comment]
>     1. What should happen with a missing version file? Treat it as an unknown and incompatible syntax/theme set?

Yes. We would assume that these are artifacts from a version of `bat` that did not have this feature and simply declare these files as incompatible.


> I don't think deleting it automatically would the correct course of action. If users update bat and find their theme or syntax mappings missing without any explanation, they might consider it a bug.
> 
> Ruling that out would leave us with two options: Warn the user and proceed anyways; or print an error and exit.

A third option might be to regenerate the files from the config folder "on the fly". Now this would of course only work if (1) the orignal path with themes and syntaxes still exists in unmodified form and (2) if we know that path.

Both points might be solved by storing additional information in the metadata file. It might be as simple as storing a file path and a hash (across all files in the `syntaxes` and `themes` folders):
```yaml
version: "7.4.0"
source:
  path: /home/shark/.config/bat
  hash: "9333753eab96a4daca9a1f69bb8a15ef"
```
